src/memory_prefetch.c

```diff
@@ -279,3 +279,30 @@ void removeClientFromPendingCommandsBatch(client *c) {
         }
     }
 }
+
+/* A simple way to prefetch the keys for a client command execution. This can be
+ * used for optimizing multi-key commands like mget, mset, etc. */
+void prefetchKeys(client *c, int first, int step, int count) {
+    /* Skip this for IO threads. Keys are already batch-prefetched. */
+    if (c->io_parsed_cmd) return;
+
+    int n = (c->argc - first) / step;
+    if (count < n) n = count;
+    if (n < 2) return; /* There's no benefit in prefetcing only one key. */
+    hashtableIncrementalFindState states[n];
+    int i, arg;
+    for (i = 0, arg = first; i < n && arg < c->argc; i++, arg += step) {
+        sds key = c->argv[arg]->ptr;
+        int slot = server.cluster_enabled ? getKeySlot(key) : 0;
```

Comment:
Isn't the slot when cluster mode is disabled be -1?

Reply:
getKVStoreIndexForKey is static in db.c so either we make it non-static. Or I can just rename `slot` to `kv_index`?